src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.internet.shop.service.impl;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.OrderService;
+import java.util.List;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order(shoppingCart.getUserId());
+        order.setProducts(List.copyOf(shoppingCart.getProducts()));
+        shoppingCartDao.clear(shoppingCart);
```

Comment:
We have to update ShoppingCart after changing

Reply:
Ok, I have refactored clearing logic